packages/flutter_tools/lib/src/build_system/targets/icon_tree_shaker.dart

```diff
@@ -0,0 +1,366 @@
+      _artifacts.getArtifactPath(Artifact.constFinder),
+    );
+    final File dart = _fs.file(
+      _artifacts.getArtifactPath(Artifact.engineDartBinary),
+    );
+
+    final Map<String, List<int>> iconData = await _findConstants(
+      dart,
+      constFinder,
+      appDill,
+    );
+    final Set<String> familyKeys = iconData.keys.toSet();
+
+    final Map<String, String> fonts = await _parseFontJson(
+      _fontManifest,
+      familyKeys,
+    );
+
+    if (fonts.length != iconData.length) {
+      throwToolExit('Expected to find fonts for ${iconData.keys}, but found '
+                    '${fonts.keys}. This usually means you are refering to '
+                    'font families in an IconData class but not including them '
+                    'in the assets section of your pubspec.yaml, are missing '
+                    'the package that would include them, or are missing '
+                    '"uses-material-design: true".');
+    }
+
+    final Map<String, _IconTreeShakerData> result = <String, _IconTreeShakerData>{};
+    for (final MapEntry<String, String> entry in fonts.entries) {
+      result[entry.value] = _IconTreeShakerData(
+        family: entry.key,
+        relativePath: entry.value,
+        codePoints: iconData[entry.key],
+      );
+    }
+    return result;
+  }
+
+  /// Calls font-subset, which transforms the `inputPath` font file to a
+  /// subsetted version at `outputPath`.
+  ///
+  /// The `relativePath` parameter
+  ///
+  /// All parameters are required.
+  ///
+  /// If [enabled] is false, or the relative path is not recognized as an icon
+  /// font used in the Flutter application, this returns false.
+  /// If the font-subset subprocess fails, it will [throwToolExit].
+  /// Otherwise, it will return true.
+  Future<bool> subsetFont({
+    @required String inputPath,
+    @required String outputPath,
+    @required String relativePath,
+  }) async {
+    if (!enabled) {
+      return false;
+    }
+
+    _iconData ??= await _getIconData(_environment);
+    assert(_iconData != null);
+
+    final _IconTreeShakerData iconTreeShakerData = _iconData[relativePath];
+    if (iconTreeShakerData == null) {
+      return false;
+    }
+
+    final File fontSubset = _fs.file(
+      _artifacts.getArtifactPath(Artifact.fontSubset),
+    );
+    if (!fontSubset.existsSync()) {
+      throwToolExit('The font-subset utility is missing. Run "flutter doctor".');
+    }
+
+    final List<String> cmd = <String>[
+      fontSubset.path,
+      outputPath,
+      inputPath,
+    ];
+    final String codePoints = iconTreeShakerData.codePoints.join(' ');
+    _logger.printTrace('Running font-subset: ${cmd.join(' ')}, '
+                       'using codepoints $codePoints');
+    final Process fontSubsetProcess = await _processManager.start(cmd);
+    try {
+      fontSubsetProcess.stdin.writeln(codePoints);
+      await fontSubsetProcess.stdin.flush();
+      await fontSubsetProcess.stdin.close();
+    } catch (e) {
```

Comment:
nit: catch the specific error/exception types you expect. As written this will catch `NoSuchMethodError` or `TypeError`

Reply:
It's not clear to me which exceptions these methods can throw.  The idea is that `exitCode` will handle it.  @zanderso is there some exception or set of exceptions to do?  `OSError` and some others?

Reply:
Catching `NoSuchMethodError` or `TypeError` is bad because those are programming errors. There are several errors that could happen writing to stdin. They'll all be derived from `Exception`, which doesn't include the `Error` types like `TypeError` etc.

Reply:
Ok.  I have a block for `on Exception` and a block for `on OSError` that I think shoudl cover us.